### DISCUSSION TITLE
Avalonia 11 is showing worse performance than 0.10 after migration

### DISCUSSION THREAD

**Discussion by Tetedeiench** [AvaloniaUI/Avalonia#12561] [Q&A]
Hi everyone,

I was redirected here from the gitter chat.

I could finally port my application to Avalonia 11, and while it was a tough and arduous task full of pitfalls, I could finally finalize the port. Took two solid days.

Despite reading that a great focus was put onto performance, my application is showing worse performance on v11 than 0.10 : CPU Usage is higher, Memory consumption is higher, all the figures are in the red sadly.

Using DotTrace to track the execution of my app, using the very same conditions ( window size, state, etc) :
 * 0.10 : 2.7% CPU, 1.5GB memory allocations (total)
 * 11 : 3.4% CPU, 2GB Memory allocations (total)
 
I must state that the scenario used was identical between 0.10 and 11, and the app state was also exactly the same (same window size, position, clicks, duration, same everything). 

Only the GUI portion of the application was changed and ported to 11 between the two measures, done in Release mode.

Here are graphs showing the CPU Usage of my application when the affinity is forced to a single core : 0.10 wins by quite a margin.
v0.10 :  ![v0 10](https://github.com/AvaloniaUI/Avalonia/assets/23140774/a4ddea77-67cc-4f51-8fdb-2896265fed84) / v11 : ![v11](https://github.com/AvaloniaUI/Avalonia/assets/23140774/c0c801db-def1-4361-afb9-f6548c10fc9f)

For Memory, measured using DotMemory : 
* 0.10 : 450.3MB
* 0.11 : 495.2MB

A rough 10% increase.

Am I the only one affected by this ? I am debating doing some analysis on a simpler UI than mine, to get more precise measures. 

Doing this isn't meaningful if this is intended or already known - I could not find any valid information regarding this though.

So, is there anyone having the same results as mine ?

**Comment by timunie** [comment]
@Tetedeiench I talked to the team and we are not really aware of such an issues. So if you can file a sample some day after your vacation that may be helpful for you and us. 

Thank you.
